src/node/interfaces.cpp

```diff
@@ -861,6 +861,25 @@ class MinerImpl : public Mining
         return tip->GetBlockHash();
     }
 
+    std::pair<uint256, int> waitTipChanged(MillisecondsDouble timeout) override
+    {
+        uint256 previous_hash{WITH_LOCK(::cs_main, return chainman().ActiveChain().Tip()->GetBlockHash();)};
+
+        auto deadline = std::chrono::steady_clock::now() + timeout;
+        {
+            WAIT_LOCK(g_best_block_mutex, lock);
+            while (!chainman().m_interrupt && std::chrono::steady_clock::now() < deadline) {
+                auto check_time = std::chrono::steady_clock::now() + std::min(timeout, MillisecondsDouble(1000));
+                g_best_block_cv.wait_until(lock, check_time);
+                if (g_best_block != previous_hash) break;
```

Comment:
In commit "Add waitTipChanged to Mining interface" (4554b3b1adf678e17d479d72738f4f2089971c4a)

If you adopt suggestion above for caller to pass in the previous_hash value, you will want to delete this line and add `g_best_block == previous_hash &&` to the while loop condition so this function does not block if the hash is already different when it is called.

Reply:
Done